app/src/main/java/movierating/services/UserServiceImpl1.java

```diff
@@ -12,7 +13,9 @@ public UserServiceImpl1(UserRepository userRepository) {
     }
 
     @Override
-    public void addUser(User user) {
+    public void addUser(User user) throws AddUserException {
+        if (userRepository.getUsers().containsValue(user))
+            throw new AddUserException("He is already a user of this Service" + user.getUserName());
```

Comment:
I think the exception message can be present inside the class of `AddUserException` itself and you just pass the user.getUserName() as argument to constructor.
Reason: Separation of concerns

Reply:
Changed this